src/interactiveHandler.py

```diff
@@ -105,15 +105,18 @@ def execute(self):
 
 class getHKTransaction(baseTransaction):
     def execute(self):
-        print("WARN: I have no idea how HK receive works, it may not at all")
         self.send()
+        rxlist = list();
         rxData = dict()
         while True:
-            ret = self.receive()
-            rxData = {**rxData, **self.parseReturnValue(ret)}
+            try:
+                ret = self.receive()
+            except:
+                return rxlist
+            rxData = self.parseReturnValue(ret)
```

Comment:
```suggestion
            rxData = self.parseReturnValue(ret)
            rxlist.append(rxData)
```
`rxlist` wasn't being modified/extended anywhere before it is returned.